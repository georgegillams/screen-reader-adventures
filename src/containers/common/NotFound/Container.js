import React from 'react';
import PropTypes from 'prop-types';
import { cssModules } from 'gg-components/helpers/cssModules';
import { Subsection } from 'gg-components/Subsection';
import { Paragraph } from 'gg-components/Paragraph';
import PageTitle from 'components/common/PageTitle';
import TextLink from 'components/common/TextLink';

import STYLES from './not-found.scss';

const getClassName = cssModules(STYLES);

const NotFound = props => {
  const { className } = props;
  const classNameFinal = [getClassName('not-found__container')];
  if (className) {
    classNameFinal.push(className);
  }

  return (
    <div className={classNameFinal.join(' ')}>
      <PageTitle className={getClassName('not-found__container')} name="Oops." pageTitle="404">
        <Subsection anchor={false}>
          <Paragraph>
            The page you&apos;re looking for doesn&apos;t exist, or you don&apos;t have permission to view it.
            <br />
            Maybe the <TextLink href={'/sitemap'}>site map</TextLink> can help
          </Paragraph>
        </Subsection>
      </PageTitle>
    </div>
  );
};

NotFound.propTypes = {
  className: PropTypes.string,
};

NotFound.defaultProps = {
  className: null,
};

export default NotFound;
