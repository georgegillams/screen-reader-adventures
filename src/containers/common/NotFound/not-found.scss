@import '~utils/tokens';

.not-found {
  &__container {
    @include centred;
  }
}
