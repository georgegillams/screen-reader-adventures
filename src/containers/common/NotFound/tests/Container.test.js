import React from 'react';
import { render } from '@testing-library/react';

import NotFound from '../Container';

describe('<NotFound />', () => {
  it('should render correctly', () => {
    const { container } = render(<NotFound />);

    expect(container).toMatchSnapshot();
  });
});
