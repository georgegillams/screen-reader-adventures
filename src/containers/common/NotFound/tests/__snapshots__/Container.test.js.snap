// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should render correctly 1`] = `
<div>
  <div
    class="not-found__container"
  >
    <h1
      class="page-title__heading not-found__container"
    >
      Oops.
    </h1>
    <div
      class="subsection__outer"
    >
      <span
        class="paragraph"
      >
        The page you're looking for doesn't exist, or you don't have permission to view it.
        <br />
        Maybe the 
        <a
          class="text-link text-link--text-link text-link--no-padding text-link--inline"
          href="/sitemap"
        >
          site map
        </a>
         can help
      </span>
    </div>
  </div>
</div>
`;
